client/descriptions.go

```diff
@@ -41,12 +43,21 @@ func (col CollectionDescription) IDString() string {
 
 // GetFieldByID searches for a field with the given ID. If such a field is found it
 // will return it and true, if it is not found it will return false.
-func (col CollectionDescription) GetFieldByID(id FieldID) (FieldDescription, bool) {
-	if !col.Schema.IsEmpty() {
-		for _, field := range col.Schema.Fields {
-			if field.ID == id {
-				return field, true
-			}
+func (col CollectionDescription) GetFieldByID(id FieldID, schema *SchemaDescription) (FieldDescription, bool) {
```

Comment:
question: This and the couple other methods bellow no longer need the `CollectionDescription`. Do you intend on turning those into methods of `SchemaDescription` in the future?

Reply:
No, this and the others remain here as whilst they currently only access stuff on the schema, that is because the stuff they access is incorrectly on the schema.

For example, in this case we are getting fields by their ID, FieldID is a local concept that should exist only on collection, not on schema.

The function is located in the correct place, but for historical reasons the data structures are incorrect and need to be changed along with the implementation.

Reply:
Thanks for clarifying